test/test_irods_http_api.py

```diff
@@ -3339,6 +3343,246 @@ def test_report_operation(self):
+            })
+            self.logger.debug(r.content)
+            self.assertEqual(r.status_code, 200)
+
+            result = r.json()
+            self.assertEqual(result['irods_response']['status_code'], 0)
+            self.assertEqual(result['exists'], True)
+            original_conn_info = result['info']['connection_info']
+
+            # The properties to update.
+            value_error_map = [
+                ('example.org:', irods_error_codes.CAT_INVALID_ARGUMENT),
+                (':1247', irods_error_codes.CAT_HOSTNAME_INVALID),
+                (':', irods_error_codes.CAT_HOSTNAME_INVALID),
+                ('', irods_error_codes.CAT_INVALID_ARGUMENT)
+            ]
+
+            # Attempt to change the properties of the new zone.
+            for conn_info, ec in value_error_map:
+                with self.subTest(f'Invalid connection info [{conn_info}] results in error code [{ec}]'):
+                    r = requests.post(self.url_endpoint, headers=headers, data={
+                        'op': 'modify',
+                        'name': zone_name,
+                        'property': 'connection_info',
+                        'value': conn_info 
+                    })
+                    self.logger.debug(r.content)
+                    self.assertEqual(r.status_code, 200)
+                    self.assertEqual(r.json()['irods_response']['status_code'], ec)
+
+                    # Show the new zone was not changed.
+                    r = requests.get(self.url_endpoint, headers=headers, params={
+                        'op': 'stat',
+                        'name': zone_name
+                    })
+                    self.logger.debug(r.content)
+                    self.assertEqual(r.status_code, 200)
+
+                    result = r.json()
+                    self.assertEqual(result['irods_response']['status_code'], 0)
+                    self.assertEqual(result['exists'], True)
+                    self.assertEqual(result['info']['connection_info'], original_conn_info)
+
+        finally:
+            # Remove the remote zone.
+            r = requests.post(self.url_endpoint, headers=headers, data={
+                'op': 'remove',
+                'name': zone_name
+            })
+            self.logger.debug(r.content)
+
+    def test_modifying_permissions_of_zone_collection(self):
+        rodsadmin_headers = {'Authorization': 'Bearer ' + self.rodsadmin_bearer_token}
+        rodsuser_headers = {'Authorization': 'Bearer ' + self.rodsuser_bearer_token}
+
+        # Add a remote zone to the local zone.
+        zone_name = 'tempZone'
+        #zone_name = 'other_zone'
+        #r = requests.post(self.url_endpoint, headers=rodsadmin_headers, data={
+        #    'op': 'add',
+        #    'name': zone_name
+        #})
+        #self.logger.debug(r.content)
+        #self.assertEqual(r.status_code, 200)
+        #self.assertEqual(r.json()['irods_response']['status_code'], 0)
```

Comment:
Does `modzonecollacl` require a remote zone? The docs kind of imply it does, but testing with `iadmin` and using the zone administration lib seem to not enforce that.

Need to test the iRODS API more.

Reply:
The local zone is adequate for this test. Remote zones aren't required to prove correctness.

This was determined through local bench testing.

Reply:
On second thought, more investigation of `iadmin modzonecollacl` is needed before making this operation available.

Disabling the operation and tests until further notice.

I'll open an issue to revisit it later. No need to hold up the PR for this one operation.